mirgecom/exceptions.py

```diff
@@ -0,0 +1,45 @@
+"""Provide custom exceptions for use in callback routines.
```

Comment:
@MTCam I had originally intended for these exceptions to be caught in a manner similar to what was originally in https://github.com/illinois-ceesd/mirgecom/pull/370/, but I think we decided it was not needed? So do you think we need these?

Reply:
I think that error-specific exceptions have a place - I'm just not sure where in the current approach. We do need to handle some things collectively before the code punches out for errors that we control (i.e. not those tossed by the dependent-packages). 

Can we give the drivers another collective go-over/review and then discuss this again?

Reply:
@MTCam I think we should just remove `mirgecom/exceptions.py`. It's easy enough to add them in later if we want custom exceptions. But at the moment none of the callback stuff is designed to catch errors; they currently raise a `RuntimeError` when something goes wrong, which I think is fine.

Alternatively, if we really want to keep them, we should at least use them. I think deleting them for now is simpler :)